### PR TITLE
abocquet/test jit sesolve

### PR DESCRIPTION
**Pull request by abocquet**
Not very critical in terms of jitting (expect for `_PWCFactor`) actually but it does fix a couple of bugs

### CODE REVIEW

**Review by gautierronan — CHANGES_REQUESTED** [review]

File: dynamiqs/time_array.py [hunk 1]
```diff
@@ -376,12 +379,16 @@ def conj(self) -> _Factor:
...
+        idx = jnp.searchsorted(self.times, t, side='right') - 1
+        return jax.lax.select(
+            t < self.times[0],
+            jnp.zeros_like(self.values[..., 0]),
+            jax.lax.select(
+                t >= self.times[-1],
+                jnp.zeros_like(self.values[..., 0]),
+                self.values[..., idx],
+            ),
+        )
```
Comment (pierreguilmin) [inline dynamiqs/time_array.py:391]: I think https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.cond.html#jax.lax.cond is better suited here.
> In general [select()](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.select.html#jax.lax.select) leads to evaluation of both branches, although the compiler may elide computations if possible. For a similar function that usually evaluates only a single branch, see [cond()](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.cond.html#jax.lax.cond).

Comment (pierreguilmin) [reply dynamiqs/time_array.py:391]: Also there's no need to nest select, and `jnp.searchsorted` should only be called in the second branch (as in the original Python code).

File: tests/solver_tester.py [hunk 2]
```diff
@@ -136,6 +138,42 @@ def _test_correctness(
...
+    def _test_jit(
+        self,
+        system: System,
+        solver: Solver,
+        *,
+        options: dict[str, Any] | None = None,
+        ntsave: int = 11,
+        ysave_atol: float = 1e-3,
+        esave_rtol: float = 1e-3,
+        esave_atol: float = 1e-5,
```
Comment (pierreguilmin) [inline tests/solver_tester.py:150]: You can remove this.

**Review by pierreguilmin — DISMISSED** [review]
A few comments but I don't want to block if you want to move fwd.

File: dynamiqs/time_array.py (hunk 1 above)
Comment (gautierronan) [reply dynamiqs/time_array.py:391]: Adressed in https://github.com/dynamiqs/dynamiqs/pull/411

**Review by pierreguilmin — APPROVED** [review]

**Review by gautierronan — APPROVED** [review]

**Pull request merged**